packages/data-service-generator/src/server/custom-module/create-custom-module.ts

```diff
@@ -0,0 +1,119 @@
+import { plural } from "pluralize";
+import { camelCase } from "camel-case";
+import { ModuleContainer, ModuleMap } from "@amplication/code-gen-types";
+import {
+  createServiceId,
+  createServiceModules,
+} from "./custom-service/create-service";
+import { createCustomModuleControllerModules } from "./custom-controller/create-controller";
+import { createCustomModuleResolverModules } from "./custom-resolver/create-resolver";
+import DsgContext from "../../dsg-context";
+import { createCustomModule } from "./custom-module/create-module";
+
+export async function createCustomModulesModules(
+  dtoNameToPath: Record<string, string>
+): Promise<ModuleMap> {
+  const context = DsgContext.getInstance;
+  const { moduleContainers } = context;
+
+  const customModuleContainers = moduleContainers.filter(
+    (module) => module.entityId === undefined
+  );
+
+  const customModules = new ModuleMap(DsgContext.getInstance.logger);
+  for await (const customModule of customModuleContainers) {
+    await customModules.merge(
+      await createCustomModuleModules(customModule, dtoNameToPath)
+    );
+  }
+
+  return customModules;
+}
+
+async function createCustomModuleModules(
+  customModule: ModuleContainer,
+  dtoNameToPath: Record<string, string>
+): Promise<ModuleMap> {
+  const customModuleName = customModule.name;
+  const context = DsgContext.getInstance;
+  const { appInfo } = context;
+
+  //validateEntityName(customModule.name); todo: validate module name
+
+  await context.logger.info(`Creating ${customModuleName}...`);
+  const resource = camelCase(plural(camelCase(customModuleName)));
+  const serviceId = createServiceId(customModuleName);
+
+  const serviceModules = await createServiceModules(
+    customModuleName,
+    serviceId,
+    dtoNameToPath
+  );
+
+  const [serviceModule] = serviceModules.modules();
+
+  const controllerModules =
+    (appInfo.settings.serverSettings.generateRestApi &&
+      (await createCustomModuleControllerModules(
+        resource,
+        customModuleName,
+        serviceModule.path,
+        dtoNameToPath
+      ))) ||
+    new ModuleMap(DsgContext.getInstance.logger);
+
+  const [controllerModule] = controllerModules.modules();
```

Comment:
if `generateRestApi` is false, and there is an empty module map here,  wouldn't this line throw? or will you just get an `undefined` value as `controllerModule` ?

Reply:
@mulygottlieb  will you just get an undefined value as controllerModule  (same logic as in the existing controller creation)